CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -902,6 +902,7 @@ protected static function isComplete($recurringContributionID, $installments) {
   public static function calculateRecurLineItems($recurId, $total_amount, $financial_type_id) {
     $originalContributionID = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $recurId, 'id', 'contribution_recur_id');
     $lineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($originalContributionID);
+    $lineSets = array();
     if (count($lineItems) == 1) {
```

Comment:
@monishdeb - Because that doesn't work!! 
when == 1 we can do calculations for a change in $total_amount; if > 1 we have no way of telling which of the lineItems may have changed - so must just pass on the original ones. That recalculation will not work for > 1 - so keeping that for == 1 only.